File: containsDuplicate.py
class Solution:
    def bruteForce(self, nums):
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] == nums[j]:
                    return True
        return False
    
    def sorting(self, nums):
        nums.sort()
        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                return True
        return False
    
    def hashing(self, nums):
        hash = set()
        for n in nums:
            if n in hash:
                return True
            hash.add(n)
        return False
    

File: containsDuplicate.ts
class Solution {
    bruteForce(nums: number[]): boolean {
        for (let i = 0; i < nums.length; i++) {
            for (let j = i + 1; j < nums.length; j++) {
                if (nums[i] === nums[j]) {
                    return true;
                }
            }
        }
        return false;
    }

    sorting(nums: number[]): boolean {
        nums.sort((a, b) => a - b);
        for (let i = 0; i < nums.length - 1; i++) {
            if (nums[i] === nums[i + 1]) {
                return true;
            }
        }
        return false;
    }

    hashing(nums: number[]): boolean {
        const hash = new Set<number>();
        for (const n of nums) {
            if (hash.has(n)) {
                return true;
            }
            hash.add(n);
        }
        return false;
    }
};

export default new Solution();
